code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -844,8 +844,9 @@
 	to_chat(src, "You are also capable of hacking APCs, which grants you more points to spend on your Malfunction powers. The drawback is that a hacked APC will give you away if spotted by the crew. Hacking an APC takes 60 seconds.")
 	view_core() //A BYOND bug requires you to be viewing your core before your verbs update
 	malf_picker = new /datum/module_picker
-	modules_action = new(malf_picker)
-	modules_action.Grant(src)
+	if(!IS_MALF_AI(src)) //antagonists have their modules built into their antag info panel. this is for adminbus and the combat upgrade
+		modules_action = new(malf_picker)
+		modules_action.Grant(src)
```

Comment:
What is this gonna do to combat upgraded AIs?

Reply:
Combat upgraded ais play the exact same as before, they just get their little funny action button. It's the traitor ais that get it moved into a tab, which is much noicer